changelogs/unreleased/gh-3159-recovery-triggers.md

```diff
@@ -0,0 +1,7 @@
+## feature/box
+
+* It is now possible to register triggers for various recovery stages: use
+  `box.ctl.on_recovery_state()` before initial `box.cfg()` call. The trigger
```

Comment:
I can't find a design document for this feature. The closest thing I see is [replication_connect_quorum rework (RCQ)](https://www.notion.so/tarantool/replication_connect_quorum-rework-RCQ-a749e1e6b8934f3388b2176ab52c4300) doc, but there this new API is just mentioned, no detailed description provided. Besides, the function is called `on_recovery_stage` there... Since this is a part of the public API, please get an approval from the PM team regarding the names and behavior of this new feature.

Reply:
Good point, I did get a verbal approval, just couldn't summon the products here.
@Mons, do you hear me?

Reply:
If @Mons could read the PR description and approve the PR if everything looks good to him, that'd be great.

In future, please describe all aspects of the proposed public API in details in the design doc so that the reviewer has no questions.

Reply:
> In future, please describe all aspects of the proposed public API in details in the design doc so that the reviewer has no questions.

Ok